### PR TITLE
Upgrade eslint to 4.x

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: chat-plugins/info.js [hunk 1]
```diff
@@ -1791,7 +1791,7 @@ exports.commands = {
 	'!dice': true,
 	roll: 'dice',
 	dice: function (target, room, user) {
-		if (!target || target.match(/[^d\d\s\-\+HL]/i)) return this.parse('/help dice');
+		if (!target || target.match(/[^d\d\s-+HL]/i)) return this.parse('/help dice');
```
Comment (Zarel) [inline chat-plugins/info.js]: Aha. The `\-` is necessary.

Comment (Zarel) [inline chat-plugins/info.js]: Personally, I would write this

`[^dHL\d\s+-]`

File: chat-plugins/info.js [hunk 2]
```diff
@@ -1808,7 +1808,7 @@ exports.commands = {
 		let offset = 0;
 		let removeOutlier = 0;
 
-		let modifierData = target.match(/[\-\+]/);
+		let modifierData = target.match(/[-+]/);
```
Comment (Zarel) [inline chat-plugins/info.js]: I would write this `[+-]`

File: chat-plugins/info.js (hunk 1 above)
Comment (bumbadadabum) [inline chat-plugins/info.js]: Is it really? Not according to eslint or regexr.com

File: chat-plugins/info.js (hunk 2 above)
Comment (bumbadadabum) [inline chat-plugins/info.js]: Want me to put in the 10 seconds of INCREDIBLY INTENSIVE LABOR to change that?

File: chat.js [hunk 3]
```diff
@@ -137,7 +137,7 @@ class PatternTester {
 	register(...elems) {
 		for (let elem of elems) {
 			this.elements.push(elem);
-			if (/^[^ \^\$\?\|\(\)\[\]]+ $/.test(elem)) {
+			if (/^[^ ^$?|()[\]]+ $/.test(elem)) {
```
Comment (Zarel) [inline chat.js]: Does it actually complain about `\[`? Ugh. I'd put it back unless it forced me to take it out.

File: chat-plugins/info.js (hunk 1 above)
Comment (Zarel) [inline chat-plugins/info.js]: Oh, I'd still put the `-` on the end for readability.

Comment (Zarel) [inline chat-plugins/info.js]: And the letters next to each other.

File: chat.js (hunk 3 above)
Comment (bumbadadabum) [inline chat.js]: it kinda does force you to take it out lol

File: chat-plugins/info.js (hunk 2 above)
Comment (Zarel) [inline chat-plugins/info.js]: yespls <3

File: chat.js (hunk 3 above)
Comment (bumbadadabum) [inline chat.js]: It's 2 cases in the entire codebase I don't think it's a big deal to just listen to eslint here.

File: chat-plugins/info.js (hunk 2 above)
Comment (Zarel) [inline chat-plugins/info.js]: In general, I prefer `[` blocks in regexes that end with `-` to ones that start with `-`, and we should enforce that in code.

File: chat-plugins/info.js (hunk 1 above)
Comment (bumbadadabum) [inline chat-plugins/info.js]: or testing it on my localhost

File: chat.js (hunk 3 above)
Comment (Zarel) [inline chat.js]: Fine... :/

File: chat-commands.js [hunk 4]
```diff
@@ -895,7 +895,7 @@ exports.commands = {
 		if (!target) return;
 		if (!/</.test(target)) {
 			// not HTML, do some simple URL linking
-			let re = /(https?:\/\/(([-\w\.]+)+(:\d+)?(\/([\w/_\.]*(\?\S+)?)?)?))/g;
+			let re = /(https?:\/\/(([-\w.]+)+(:\d+)?(\/([\w/_.]*(\?\S+)?)?)?))/g;
```
Comment (Zarel) [inline chat-commands.js]: Here's another block where I'd prefer to end with `-]` rather than starting with `[`

File: chat-commands.js [hunk 5]
```diff
@@ -950,7 +950,7 @@ exports.commands = {
...
```
Comment (Zarel) [inline chat-commands.js]: Ditto

**Pull request merged**